docs/administration-guide/gaffer-deployment/gremlin.md

```diff
@@ -0,0 +1,164 @@
+
+```properties
+# The Tinkerpop graph class we should use
+gremlin.graph=uk.gov.gchq.gaffer.tinkerpop.GafferPopGraph
+gaffer.graphId=graphProxy
+gaffer.storeproperties=conf/gaffer/store.properties
+gaffer.userId=user01
+```
+
+The second file needed is the configuration for the Gremlin server, this is
+what ties everything together and makes sure the server runs using the Gaffer
+implementation we have configured. A default file is provided in the
+[Tinkerpop repository](https://github.com/apache/tinkerpop/blob/master/gremlin-server/conf/gremlin-server.yaml).
+
+From this file two places need modifying, the first is to change it to use
+our graph configuration file by modifying the `graphs` section like so:
+
+```yaml
+graphs: {
+  graph: conf/gafferpop/gafferpop.properties
+}
+```
+
+The second place is to ensure the Gaffer plugin is loaded for Gremlin which is
+achieved by adding the following to the list of plugins in the `plugins`
+section:
+
+```yaml
+uk.gov.gchq.gaffer.tinkerpop.gremlinplugin.GafferPopGremlinPlugin: {}
+```
+
+!!! tip
+    See the [Tinkerpop docs](https://tinkerpop.apache.org/docs/current/reference/#gremlin-server)
+    for more information on Gremlin server configuration.
+
+### Running the Gremlin Server
+
+After following the previous steps you should now have three custom files
+created which we will bind mount into a `gremlin-server` container. One final
+step however, is to obtain required Gaffer JARs and add them to the container as
+well, there are many different ways to do this the easiest being through maven
+which can use following goal to download all dependencies from a POM:
+
+```bash
+mvn clean dependency:copy-dependencies
+```
+
+Once all JARs are available they can be bind mounted to a path such as,
+`ext/gafferpop/plugin/` in the container to be added to the classpath.
+
+The bind mount location of the custom configuration files are as follows:
+
+- `store.properties` -> `conf/gaffer/store.properties`
+- `gafferpop.properties` -> `conf/gafferpop/gafferpop.properties`
+- `gremlin-server.yaml` -> `conf/gremlin-server.yaml`
+
+The container can then be ran as normal with the above bind mounts and
+specifying the `conf/gremlin-server.yaml` file as the run argument for the
+container, for example:
+
+```bash
+docker run \
+       --detach \
+       --name gaffer-gremlin \
+       --hostname gaffer-gremlin \
+       --publish 8182:8182 \
+       --net host \
+       --volume store.properties:conf/gaffer/store.properties \
+       --volume gafferpop.properties:conf/gafferpop/gafferpop.properties \
+       --volume gremlin-server.yaml:conf/gremlin-server.yaml \
+       tinkerpop/gremlin-server:latest gremlin-server.yaml
+```
+
+!!! tip
+    You can also use a different Gaffer store type other than a Proxy Store if
+    you want to connect directly to an Accumulo instance; however, for this you
+    will need to know the graph Schema and other configuration options.
```

Comment:
This sentence could maybe a little clearer and potentially would be better as two sentences.
```suggestion
    You can also use a different Gaffer store type other than a Proxy Store, but this can be more complex to set up. For example, if you want to connect directly to an Accumulo instance you
    will need to know the graph Schema and other configuration options.
```